homeassistant/components/stream/worker.py

```diff
@@ -119,33 +144,71 @@ def mux_packet(self, packet):
             packet.stream = self._output_audio_stream
             self._av_output.mux(packet)
 
-    def flush(self, duration):
+    def check_flush_part(self, packet: av.Packet) -> None:
+        """Check for and mark a part segment boundary and record its duration."""
+        byte_position = self._memory_file.tell()
+        if self._segment.last_write_pos == byte_position:
+            return
+        if self._segment.last_write_pos == 0:  # The beginning of the first moof
```

Comment:
That sounds good

Reply:
I tried this but the downside is it requires adding two additional expressions in the first quick return conditional (because the `Segment` doesn't exist and its `last_write_pos` member isn't initialized to 0). One way around this is to move `last_write_pos` from a member on `Segment` to a member on `SegmentBuffer`. I'll try this and see how it looks.
We only use it outside the worker in one place in `hls.py`, but that can be easily replaced by a call to `len()`